pkg/operator/reconciler.go

```diff
@@ -0,0 +1,133 @@
+)
+
+const ClusterOperatorName = "node-tuning"
+
+// Reconciler reconciles a Tuned object and creates the tuned daemon set
+type Reconciler struct {
+	client.Client
+	Scheme    *runtime.Scheme
+	Recorder  record.EventRecorder
+	AssetsDir string
+}
+
+// SetupWithManager creates a new PerformanceProfile Controller and adds it to the Manager.
+// The Manager will set fields on the Controller and Start it when the Manager is Started.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// we want to initiate reconcile loop only on changes under the NTO cluster operator
+	p := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			if e.Object == nil {
+				klog.Error("Create event has no runtime object to create")
+				return false
+			}
+
+			if e.Object.GetName() != ClusterOperatorName {
+				return false
+			}
+
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if !validateUpdateEvent(&e) {
+				return false
+			}
+
+			if e.ObjectOld.GetName() != ClusterOperatorName {
+				return false
+			}
+
+			return true
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			if e.Object == nil {
+				klog.Error("Delete event has no runtime object to delete")
+				return false
+			}
+
+			if e.Object.GetName() != ClusterOperatorName {
+				return false
+			}
+
+			return true
+		},
+	}
+
+	err := ctrl.NewControllerManagedBy(mgr).
+		For(&configapiv1.ClusterOperator{}, builder.WithPredicates(p)).
+		Owns(&appsv1.DaemonSet{}).
+		Owns(&tunedv1.Tuned{}).
+		Complete(r)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func validateUpdateEvent(e *event.UpdateEvent) bool {
+	if e.ObjectOld == nil {
+		klog.Error("Update event has no old runtime object to update")
+		return false
+	}
+	if e.ObjectNew == nil {
+		klog.Error("Update event has no new runtime object for update")
+		return false
+	}
+
+	return true
+}
+
+// Reconcile reconciles once NTO ClusterOperator created, updated and deleted,
+// and creates NTO default tuned and tuned daemon
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// Fetch the PerformanceProfile instance
+	instance := &configapiv1.ClusterOperator{}
+	err := r.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if k8serros.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	klog.Info("Reconciling ClusterOperator %q", instance.Name)
+
+	// TODO: it unclear what actions should we do once the ClusterOperator was deleted
+	if instance.DeletionTimestamp != nil {
+		return reconcile.Result{}, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *Reconciler) syncDaemonSet() {
```

Comment:
this should be removed from the current controller and handled in the CVO controller ?